challenges/08-coding-interview-prep/rosetta-code.json

```diff
@@ -7761,6 +7761,724 @@
+          "testString":
+            "assert(typeof isLychrel(12) == 'boolean', '<code>isLychrel(12)</code> should return a boolean.');"
+        },
+        {
+          "text":
+            "assert.equal(isLychrel(12), false, '<code>isLychrel(12)</code> should return <code>false</code>.'",
+          "testString":
+            "assert.equal(isLychrel(12), false, '<code>isLychrel(12)</code> should return <code>false</code>.');"
+        },
+        {
+          "text":
+            "assert.equal(isLychrel(55), false, '<code>isLychrel(55)</code> should return <code>false</code>.'",
+          "testString":
+            "assert.equal(isLychrel(55), false, '<code>isLychrel(55)</code> should return <code>false</code>.');"
+        },
+        {
+          "text":
+            "assert.equal(isLychrel(196), true, '<code>isLychrel(196)</code> should return <code>true</code>.'",
+          "testString":
+            "assert.equal(isLychrel(196), true, '<code>isLychrel(196)</code> should return <code>true</code>.');"
+        },
+        {
+          "text":
+            "assert.equal(isLychrel(879), true, '<code>isLychrel(879)</code> should return <code>true</code>.'",
+          "testString":
+            "assert.equal(isLychrel(879), true, '<code>isLychrel(879)</code> should return <code>true</code>.');"
+        },
+        {
+          "text":
+            "assert.equal(isLychrel(44987), false, '<code>isLychrel(44987)</code> should return <code>false</code>.'",
+          "testString":
+            "assert.equal(isLychrel(44987), false, '<code>isLychrel(44987)</code> should return <code>false</code>.');"
+        },
+        {
+          "text":
+            "assert.equal(isLychrel(7059), true, '<code>isLychrel(7059)</code> should return <code>true</code>.'",
+          "testString":
+            "assert.equal(isLychrel(7059), true, '<code>isLychrel(7059)</code> should return <code>true</code>.');"
+        }
+      ],
+      "id": "5a23c84252665b21eecc7efe",
+      "challengeType": 5,
+      "releasedOn": "July 10, 2018",
+      "files": {
+        "indexjs": {
+          "key": "indexjs",
+          "ext": "js",
+          "name": "index",
+          "contents": ["function isLychrel (n) {", "  // Good luck!", "}"],
+          "head": [],
+          "tail": []
+        }
+      }
+    },
+    {
+      "title": "LZW compression",
+      "description": [
+        "The Lempel-Ziv-Welch (LZW) algorithm provides loss-less data compression.",
+        "You can read a complete description of it in the  <a href=\"https://en.wikipedia.org/wiki/Lempel-Ziv-Welch\">Wikipedia article</a>  on the subject.  It was patented, but it entered the public domain in 2004.",
+        "Write a function that takes two parameters. First parameter is a boolean. True indicates compress and false indicates decompress. Second parameter is a string in case of compression, array in case of decompression. Return an array for compression. Return a string for decompression."
```

Comment:
"Write a function that takes two parameters. The first parameter is a boolean. True indicates compress and false indicates decompress. The second parameter is either a string or an array to be compressed or decompressed, respectively. If it is a string to be compressed, return an array of numbers. If it's an array of numbers to be decompressed, return a string."